feat(lb): add troubleshooting (#4591)
scaleway/docs-content: pages/load-balancer/troubleshooting/k8s-errors.mdx

---
meta:
  title: I am experiencing problems with my Kubernetes Load Balancer
  description: Troubleshoot problems with your Kubernetes Load Balancer. Discover the solutions to common errors, and get your Load Balancer service back up and running.
content:
  h1: I am experiencing problems with my Kubernetes Load Balancer
  paragraph: Troubleshoot problems with your Kubernetes Load Balancer. Discover the solutions to common errors, and get your Load Balancer service back up and running.
tags: load-balancer kubernetes annotations
dates:
  validation: 2025-03-06
  posted: 2025-03-06
categories:
  - network
---

> If you are experiencing errors with your Kubernetes Kapsule Load Balancer. This page helps you find solutions to some of the most common problems.

Author: ```suggestion
If you are experiencing errors with your Kubernetes Kapsule Load Balancer, this page may help you find solutions to some of the most common problems.
```

<Message type="important">

> You should **never** try to create or modify a Kubernetes Kapsule's Load Balancer via the Scaleway console, the API, or any other devtools.

Reviewer: ```suggestion
You should **never** try to create or modify a Kubernetes Kapsule's Load Balancer via the Scaleway console, the API, or any other developer tools.
```

> This leads to unexpected and unreliable behaviour, as the Kluster's **C**loud **C**ontroller **M**anager is unaware of the Load Balancer and attempts to overwrite configurations made in the console.

Author: ```suggestion
This leads to unexpected and unreliable behaviour, as the Cluster's **C**loud **C**ontroller **M**anager is unaware of the Load Balancer and attempts to overwrite configurations made in the console.
```

> Always provision and modify Kubernetes Load Balancers via the [CCM](/kubernetes/reference-content/kubernetes-load-balancer/#creating-a-load-balancer-for-your-cluster-overview). Use [annotations](/kubernetes/reference-content/kubernetes-load-balancer/#defining-your-load-balancers-configuration-via-annotations) to modify configure your cluster’s Load Balancer.

Reviewer 2: ```suggestion
Always provision and modify Kubernetes Load Balancers via the [CCM](/kubernetes/reference-content/kubernetes-load-balancer/#creating-a-load-balancer-for-your-cluster-overview). Use [annotations](/kubernetes/reference-content/kubernetes-load-balancer/#defining-your-load-balancers-configuration-via-annotations) to configure your cluster’s Load Balancer.
```

</Message>

## I'm experiencing connectivity issues with my Kubernetes Load Balancer

You may find that your Load Balancer is not connecting to nodes in your Kapsule cluster, meaning that health checks are failing and your application is inaccessible from the internet

### Cause

A configuration issue is preventing successful communication between your Load Balancer and the cluster's nodes.

### Solutions

> - Ensure that you provisioned and configured your Load Balancer via Kubernetes nad not via the Scaleway console, which provokes unexpected behaviors and errors.

Reviewer 2: ```suggestion
- Ensure that you provisioned and configured your Load Balancer via Kubernetes and not via the Scaleway console, which provokes unexpected behaviors and errors.
```
